main.rb

```diff
@@ -0,0 +1,31 @@
+MORSE_CODE = { '.-': 'a', '-...': 'b', '-.-.': 'c', '-..': 'd',
+               '.': 'e', '..-.': 'f', '--.': 'g', '....': 'h', '..': 'i',
+               '.---': 'j', '-.-': 'k', '.-..': 'l', '--': 'm',
+               '-.': 'n', '---': 'o', '.--.': 'p', '--.-': 'q',
+               '.-.': 'r', '...': 's', '-': 't', '..-': 'u',
+               '...-': 'v', '.--': 'w', '-..-': 'x', '-.--': 'y',
+               '--..': 'z', '.----': '1', '..---': '2', '...--': '3',
+               '....-': '4', '.....': '5', '-....': '6', '--...': '7',
+               '---..': '8', '----.': '9', '-----': '0', '|': ' ' }.freeze
+
+def decode_morse_code_character(incoded_character)
+  MORSE_CODE[:"#{incoded_character}"].upcase
+end
+
+def decode_morse_code_word(incoded_word)
+  word_array = incoded_word.split
+  word = ''
+  word_array.each { |c| word << decode_morse_code_character(c) }
+  word
+end
+
+def decode_morse_code_message(incoded_message)
+  message_array = incoded_message.split('   ')
+  message = ''
+  message_array.each { |word| message << " #{decode_morse_code_word(word)}" }
+  message
```

Comment:
- [ ] From the image below, it appears there is a leading space in the decoded sentences. If you are struggling with white space simply add `.strip` to the `message` in line 26 to fix it. i.e. `message.strip`

  Result | Expected result
  :---:|:---:
  ![trim](https://user-images.githubusercontent.com/54848363/178834659-0f8d0b86-a66a-4489-b720-3e9a8f847bc3.jpg) | ![trim-main](https://user-images.githubusercontent.com/54848363/178834639-badb0b67-58a7-4730-a86f-370285274128.jpg)